### PR TITLE
return empty strings in iso8601 as empty, not "now"

### PR DESCRIPTION
**Pull request by iampersistent**

### CODE REVIEW

File: src/Transformer/DateTimeIso8601Transformer.php [hunk 1]
```diff
@@ -14,15 +14,16 @@ class DateTimeIso8601Transformer extends AbstractTransformer
     public function transform(ResourceInterface $resource)
     {
         $value = $this->getPropertyValueFromResource($resource);
+        if (empty($value)) {
```
Comment (ruskiyos) [inline src/Transformer/DateTimeIso8601Transformer.php]: out of curiosity, why was this added? when passing into setters, if it's null, then a "typed" param is allowed. otherwise, if it's an '', a type error is returned.


`setDate(Carbon $date = null)`
if untransformed to `null`, this will work. if untransformed to `''`, this will fail

Comment (alexcoffee) [inline src/Transformer/DateTimeIso8601Transformer.php]: I think so that carbon parser doesn't create now(). I think it needs to return `null` like before.
